rotkehlchen/tests/unit/test_defi_oracles.py

```diff
@@ -94,6 +111,49 @@ def test_uniswap_oracles_historic_price(inquirer_defi, socket_enabled):  # pylin
         )
 
 
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+@pytest.mark.parametrize('should_mock_current_price_queries', [False])
+@pytest.mark.parametrize('base_manager_connect_at_start', [(WeightedNode(node_info=NodeName(name='llama', endpoint='https://base.llamarpc.com', owned=True, blockchain=SupportedBlockchain.BASE), active=True, weight=ONE),)])  # noqa: E501
+@pytest.mark.parametrize('polygon_pos_manager_connect_at_start', [(WeightedNode(node_info=NodeName(name='polygon', endpoint='https://polygon.drpc.org', owned=True, blockchain=SupportedBlockchain.POLYGON_POS), active=True, weight=ONE),)])  # noqa: E501
+def test_uniswap_oracles_evm(inquirer_defi: 'Inquirer') -> None:
+    """Test that Uniswap V2 and V3 oracles return correct prices in evm chains"""
+    assert inquirer_defi._uniswapv2 is not None
+    assert inquirer_defi._uniswapv3 is not None
+
+    # v2 historical on base
+    assert inquirer_defi._uniswapv2.query_historical_price(
+        from_asset=A_ETH,
+        to_asset=A_BASE_USDC,
+        timestamp=Timestamp(1725944400),
+    ) == Price(FVal('2340.829327449607'))
+
+    # v2 current on arbitrum
+    assert inquirer_defi._uniswapv2.query_current_price(
+        from_asset=A_ETH,
+        to_asset=A_ARBITRUM_USDC,
+    ) == Price(FVal('3212.0656284514484'))
+
+    # v3 historical on polygon
+    assert inquirer_defi._uniswapv3.query_historical_price(
+        from_asset=A_POLYGON_POS_MATIC,
+        to_asset=A_POLYGON_USDC,
+        timestamp=Timestamp(1725944400),
+    ) == Price(FVal('0.3785896756190174'))
+
+    # v3 current on optimism
+    assert inquirer_defi._uniswapv3.query_current_price(
+        from_asset=A_ETH,
+        to_asset=A_OPTIMISM_USDT,
+    ) == Price(FVal('3214.5282531850976'))
+
+    # v3 current on bsc
+    assert inquirer_defi._uniswapv3.query_current_price(
+        from_asset=A_BSC_USDT,
+        to_asset=A_BSC_BNB,
+    ) == Price(FVal('0.001433422881060733'))
```

Comment:
From the comment in the implementation I think we should shoudl see polygon pos matic and gnosis xDAI and BSC BSC token. To make sure native tokens are handled properly.